*** Starting uWSGI 2.0.18 (64bit) on [Thu Mar 12 16:14:44 2020] ***
compiled with version: 7.5.0 on 12 March 2020 15:24:34
os: Linux-4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019
nodename: ip-172-31-45-93
machine: x86_64
clock source: unix
detected number of CPU cores: 1
current working directory: /home/ubuntu/arbitrage
writing pidfile to uwsgi.pid
detected binary path: /home/ubuntu/.local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /home/ubuntu/arbitrage
your processes number limit is 3840
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to UNIX address /tmp/invest.sock fd 3
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
PEP 405 virtualenv detected: /home/ubuntu/arbitrage/venv
Set PythonHome to /home/ubuntu/arbitrage/venv
Python main interpreter initialized at 0x5634214f2510
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 416720 bytes (406 KB) for 8 cores
*** Operational MODE: preforking+threaded ***
Traceback (most recent call last):
  File "./main_flask.py", line 2, in <module>
    from flask import Flask, escape, request
ModuleNotFoundError: No module named 'flask'
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 21386)
spawned uWSGI worker 1 (pid: 21387, cores: 2)
spawned uWSGI worker 2 (pid: 21388, cores: 2)
spawned uWSGI worker 3 (pid: 21389, cores: 2)
spawned uWSGI worker 4 (pid: 21390, cores: 2)
...gracefully killing workers...
Gracefully killing worker 1 (pid: 21387)...
Gracefully killing worker 2 (pid: 21388)...
Gracefully killing worker 3 (pid: 21389)...
Gracefully killing worker 4 (pid: 21390)...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
worker 3 buried after 1 seconds
worker 4 buried after 1 seconds
binary reloading uWSGI...
chdir() to /home/ubuntu/arbitrage
closing all non-uwsgi socket fds > 2 (max_fd = 1024)...
found fd 3 mapped to socket 0 (/tmp/invest.sock)
running /home/ubuntu/.local/bin/uwsgi
[uWSGI] getting INI configuration from uwsgi.ini
*** Starting uWSGI 2.0.18 (64bit) on [Thu Mar 12 16:17:35 2020] ***
compiled with version: 7.5.0 on 12 March 2020 15:24:34
os: Linux-4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019
nodename: ip-172-31-45-93
machine: x86_64
clock source: unix
detected number of CPU cores: 1
current working directory: /home/ubuntu/arbitrage
detected binary path: /home/ubuntu/.local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /home/ubuntu/arbitrage
your processes number limit is 3840
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 inherited UNIX address /tmp/invest.sock fd 3
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
PEP 405 virtualenv detected: /home/ubuntu/arbitrage/
Set PythonHome to /home/ubuntu/arbitrage/
Python main interpreter initialized at 0x557b0310b430
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 416720 bytes (406 KB) for 8 cores
*** Operational MODE: preforking+threaded ***
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x557b0310b430 pid: 21386 (default app)
*** uWSGI is running in multiple interpreter mode ***
gracefully (RE)spawned uWSGI master process (pid: 21386)
spawned uWSGI worker 1 (pid: 21454, cores: 2)
spawned uWSGI worker 2 (pid: 21455, cores: 2)
spawned uWSGI worker 3 (pid: 21456, cores: 2)
spawned uWSGI worker 4 (pid: 21457, cores: 2)
[pid: 21454|app: 0|req: 2/1] 172.31.45.93 () {32 vars in 353 bytes} [Thu Mar 12 16:19:37 2020] GET /taoli => generated 2017 bytes in 21157 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21457|app: 0|req: 1/2] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:19:49 2020] GET /taoli => generated 2017 bytes in 18191 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21457|app: 0|req: 2/3] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:20:59 2020] GET /taoli => generated 2017 bytes in 15973 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21454|app: 0|req: 3/4] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:15 2020] GET /taoli => generated 2017 bytes in 29591 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21455|app: 0|req: 2/5] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:28 2020] GET /taoli => generated 2017 bytes in 17468 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21455|app: 0|req: 2/6] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:30 2020] GET /taoli => generated 2017 bytes in 16798 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:21:51 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21454|app: 0|req: 4/7] 202.184.199.34 () {46 vars in 868 bytes} [Thu Mar 12 16:19:45 2020] GET /taoli => generated 0 bytes in 125694 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
[pid: 21456|app: 0|req: 2/8] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:44 2020] GET /taoli => generated 2017 bytes in 17401 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21455|app: 0|req: 3/9] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:47 2020] GET /taoli => generated 2017 bytes in 18335 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21454|app: 0|req: 5/10] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:45 2020] GET /taoli => generated 2017 bytes in 29788 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21454|app: 0|req: 5/11] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:22:02 2020] GET /taoli => generated 2017 bytes in 22620 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:22:52 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21457|app: 0|req: 4/12] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:25 2020] GET /taoli => generated 0 bytes in 87203 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
Thu Mar 12 16:23:21 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:23:21 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21456|app: 0|req: 3/13] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:33 2020] GET /taoli => generated 0 bytes in 107502 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
[pid: 21455|app: 0|req: 4/14] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:22:53 2020] GET /taoli => generated 2017 bytes in 30096 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:23:34 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:23:34 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21457|app: 0|req: 5/15] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:21:23 2020] GET /taoli => generated 0 bytes in 130861 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
[pid: 21457|app: 0|req: 5/16] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:23:15 2020] GET /taoli => generated 2017 bytes in 22092 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21456|app: 0|req: 3/17] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:23:17 2020] GET /taoli => generated 2017 bytes in 24998 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21454|app: 0|req: 7/18] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:15 2020] GET /taoli => generated 2017 bytes in 23455 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21454|app: 0|req: 8/19] 66.102.6.144 () {38 vars in 650 bytes} [Thu Mar 12 16:24:14 2020] GET /taoli => generated 2017 bytes in 28551 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21454|app: 0|req: 9/20] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:41 2020] GET /taoli => generated 2017 bytes in 15954 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21457|app: 0|req: 7/21] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:38 2020] GET /taoli => generated 2017 bytes in 20553 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21457|app: 0|req: 7/22] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:36 2020] GET /taoli => generated 2017 bytes in 27169 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21457|app: 0|req: 8/23] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:25:04 2020] GET /taoli => generated 2017 bytes in 12810 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21455|app: 0|req: 6/24] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:42 2020] GET /taoli => generated 2017 bytes in 35284 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21454|app: 0|req: 10/25] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:46 2020] GET /taoli => generated 2017 bytes in 41311 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21456|app: 0|req: 5/26] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:25:17 2020] GET /taoli => generated 2017 bytes in 15387 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:26:02 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:26:02 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21456|app: 0|req: 5/27] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:57 2020] GET /taoli => generated 0 bytes in 65509 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
[pid: 21456|app: 0|req: 6/28] 202.184.199.34 () {44 vars in 872 bytes} [Thu Mar 12 16:26:27 2020] GET /taoli%20uwsgi%20nginx => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
Thu Mar 12 16:27:16 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21455|app: 0|req: 6/29] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:24:44 2020] GET /taoli => generated 0 bytes in 152104 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
[pid: 21454|app: 0|req: 11/30] 117.173.212.236 () {40 vars in 710 bytes} [Thu Mar 12 16:28:51 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 0)
...gracefully killing workers...
Gracefully killing worker 1 (pid: 21454)...
Gracefully killing worker 2 (pid: 21455)...
Gracefully killing worker 3 (pid: 21456)...
Gracefully killing worker 4 (pid: 21457)...
worker 2 buried after 1 seconds
worker 3 buried after 1 seconds
worker 4 buried after 1 seconds
Thu Mar 12 16:32:55 2020 - worker 1 (pid: 21454) is taking too much time to die...NO MERCY !!!
worker 1 buried after 1 seconds
binary reloading uWSGI...
chdir() to /home/ubuntu/arbitrage
closing all non-uwsgi socket fds > 2 (max_fd = 1024)...
found fd 3 mapped to socket 0 (/tmp/invest.sock)
running /home/ubuntu/.local/bin/uwsgi
[uWSGI] getting INI configuration from uwsgi.ini
*** Starting uWSGI 2.0.18 (64bit) on [Thu Mar 12 16:32:56 2020] ***
compiled with version: 7.5.0 on 12 March 2020 15:24:34
os: Linux-4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019
nodename: ip-172-31-45-93
machine: x86_64
clock source: unix
detected number of CPU cores: 1
current working directory: /home/ubuntu/arbitrage
detected binary path: /home/ubuntu/.local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /home/ubuntu/arbitrage
*** WARNING: you are running uWSGI without its master process manager ***
your processes number limit is 3840
your memory page size is 4096 bytes
 *** WARNING: you have enabled harakiri without post buffering. Slow upload could be rejected on post-unbuffered webservers *** 
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 inherited UNIX address /tmp/invest.sock fd 3
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
PEP 405 virtualenv detected: /home/ubuntu/arbitrage/
Set PythonHome to /home/ubuntu/arbitrage/
Python main interpreter initialized at 0x558b428d3560
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 240 seconds
mapped 333376 bytes (325 KB) for 8 cores
*** Operational MODE: preforking+threaded ***
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x558b428d3560 pid: 21386 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI worker 1 (pid: 21386, cores: 2)
spawned uWSGI worker 2 (pid: 21714, cores: 2)
spawned uWSGI worker 3 (pid: 21715, cores: 2)
spawned uWSGI worker 4 (pid: 21716, cores: 2)
Thu Mar 12 16:33:11 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:33:11 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21386|app: 0|req: 1/1] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:32:56 2020] GET /taoli => generated 0 bytes in 14812 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
[pid: 21386|app: 0|req: 2/2] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:33:29 2020] GET /taoli => generated 2017 bytes in 11885 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[2020-03-12 16:37:45,976] ERROR in app: Exception on /taoli [GET]
Traceback (most recent call last):
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/urllib3/response.py", line 425, in _error_catcher
    yield
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/urllib3/response.py", line 755, in read_chunked
    chunk = self._handle_chunk(amt)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/urllib3/response.py", line 708, in _handle_chunk
    returned_chunk = self._fp._safe_read(self.chunk_left)
  File "/usr/lib/python3.6/http/client.py", line 624, in _safe_read
    raise IncompleteRead(b''.join(s), amt)
http.client.IncompleteRead: IncompleteRead(5710 bytes read, 2586 more expected)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/models.py", line 751, in generate
    for chunk in self.raw.stream(chunk_size, decode_content=True):
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/urllib3/response.py", line 560, in stream
    for line in self.read_chunked(amt, decode_content=decode_content):
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/urllib3/response.py", line 781, in read_chunked
    self._original_response.close()
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/urllib3/response.py", line 443, in _error_catcher
    raise ProtocolError("Connection broken: %r" % e, e)
urllib3.exceptions.ProtocolError: ('Connection broken: IncompleteRead(5710 bytes read, 2586 more expected)', IncompleteRead(5710 bytes read, 2586 more expected))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "./main_flask.py", line 32, in taoli
    resp = s.get(URLS[name])
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/sessions.py", line 665, in send
    history = [resp for resp in gen] if allow_redirects else []
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/sessions.py", line 665, in <listcomp>
    history = [resp for resp in gen] if allow_redirects else []
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/sessions.py", line 245, in resolve_redirects
    **adapter_kwargs
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/sessions.py", line 683, in send
    r.content
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/models.py", line 829, in content
    self._content = b''.join(self.iter_content(CONTENT_CHUNK_SIZE)) or b''
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/models.py", line 754, in generate
    raise ChunkedEncodingError(e)
requests.exceptions.ChunkedEncodingError: ('Connection broken: IncompleteRead(5710 bytes read, 2586 more expected)', IncompleteRead(5710 bytes read, 2586 more expected))
Thu Mar 12 16:37:45 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:37:45 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (172.31.45.93)
OSError: write error
[pid: 21714|app: 0|req: 1/3] 172.31.45.93 () {32 vars in 353 bytes} [Thu Mar 12 16:36:43 2020] GET /taoli => generated 0 bytes in 62535 msecs (HTTP/1.1 500) 2 headers in 0 bytes (0 switches on core 0)
[pid: 21714|app: 0|req: 2/4] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:37:50 2020] GET /taoli => generated 2017 bytes in 14756 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21714|app: 0|req: 4/5] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:15 2020] GET /taoli => generated 2017 bytes in 11351 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21386|app: 0|req: 3/6] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:27 2020] GET /taoli => generated 2017 bytes in 15038 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21714|app: 0|req: 4/7] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:05 2020] GET /taoli => generated 2017 bytes in 37675 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21386|app: 0|req: 5/8] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:42 2020] GET /taoli => generated 2017 bytes in 43556 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:39:28 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21716|app: 0|req: 2/9] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:22 2020] GET /taoli => generated 0 bytes in 65709 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
Thu Mar 12 16:39:36 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21715|app: 0|req: 2/10] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:19 2020] GET /taoli => generated 0 bytes in 77877 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
[pid: 21714|app: 0|req: 6/11] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:17 2020] GET /taoli => generated 2017 bytes in 20151 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21386|app: 0|req: 6/12] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:19 2020] GET /taoli => generated 2017 bytes in 22516 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21716|app: 0|req: 3/13] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:28 2020] GET /taoli => generated 2017 bytes in 16283 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 3/14] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:36 2020] GET /taoli => generated 2017 bytes in 19486 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21386|app: 0|req: 7/15] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:41 2020] GET /taoli => generated 2017 bytes in 18041 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21716|app: 0|req: 4/16] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:44 2020] GET /taoli => generated 2017 bytes in 18457 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 4/17] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:56 2020] GET /taoli => generated 2017 bytes in 15322 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21716|app: 0|req: 5/18] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:40:02 2020] GET /taoli => generated 2017 bytes in 14976 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:40:21 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21386|app: 0|req: 8/19] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:25 2020] GET /taoli => generated 0 bytes in 56054 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
[pid: 21715|app: 0|req: 5/20] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:40:11 2020] GET /taoli => generated 2017 bytes in 11944 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21386|app: 0|req: 9/21] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:59 2020] GET /taoli => generated 2017 bytes in 27071 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
[pid: 21716|app: 0|req: 6/22] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:40:17 2020] GET /taoli => generated 2017 bytes in 16832 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 6/23] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:40:23 2020] GET /taoli => generated 2017 bytes in 14467 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21716|app: 0|req: 7/24] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:40:34 2020] GET /taoli => generated 2017 bytes in 25860 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Thu Mar 12 16:41:19 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21714|app: 0|req: 7/25] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:38:43 2020] GET /taoli => generated 0 bytes in 155821 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
Thu Mar 12 16:41:19 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:41:19 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21714|app: 0|req: 7/26] 202.184.199.34 () {44 vars in 844 bytes} [Thu Mar 12 16:39:37 2020] GET /taoli => generated 0 bytes in 101901 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
Thu Mar 12 16:41:26 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21386|app: 0|req: 10/27] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:40:21 2020] GET /taoli => generated 0 bytes in 64708 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 1)
Thu Mar 12 16:41:27 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Thu Mar 12 16:41:27 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21386|app: 0|req: 10/28] 202.184.199.34 () {46 vars in 875 bytes} [Thu Mar 12 16:40:26 2020] GET /taoli => generated 0 bytes in 60549 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
[pid: 21386|app: 0|req: 11/29] 120.235.43.151 () {40 vars in 709 bytes} [Thu Mar 12 17:27:26 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 1)
[pid: 21714|app: 0|req: 8/30] 36.4.83.116 () {40 vars in 706 bytes} [Thu Mar 12 23:29:50 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 1)
[pid: 21716|app: 0|req: 8/31] 36.4.83.116 () {40 vars in 890 bytes} [Thu Mar 12 23:31:40 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 1)
[pid: 21715|app: 0|req: 7/32] 14.116.133.171 () {46 vars in 1022 bytes} [Fri Mar 13 00:06:05 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 1)
[pid: 21714|app: 0|req: 9/33] 202.184.199.34 () {42 vars in 938 bytes} [Fri Mar 13 01:20:45 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 0)
[pid: 21386|app: 0|req: 12/34] 202.184.199.34 () {44 vars in 780 bytes} [Fri Mar 13 01:20:54 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 0)
[pid: 21386|app: 0|req: 13/35] 202.184.199.34 () {44 vars in 778 bytes} [Fri Mar 13 01:21:01 2020] GET /taoli => generated 1636 bytes in 29167 msecs (HTTP/1.1 200) 2 headers in 81 bytes (2 switches on core 1)
Fri Mar 13 01:31:34 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli (202.184.199.34)
OSError: write error
[pid: 21714|app: 0|req: 10/36] 202.184.199.34 () {46 vars in 809 bytes} [Fri Mar 13 01:30:27 2020] GET /taoli => generated 0 bytes in 67130 msecs (HTTP/1.1 200) 2 headers in 0 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 8/37] 202.184.199.34 () {46 vars in 809 bytes} [Fri Mar 13 01:31:31 2020] GET /taoli => generated 2926 bytes in 44175 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 9/38] 202.184.199.34 () {46 vars in 870 bytes} [Fri Mar 13 02:53:08 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 10/39] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:53:15 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 11/40] 202.184.199.34 () {44 vars in 846 bytes} [Fri Mar 13 02:53:46 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 12/41] 202.184.199.34 () {44 vars in 846 bytes} [Fri Mar 13 02:53:55 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
Gracefully killing worker 1 (pid: 21386)...
[pid: 21715|app: 0|req: 13/42] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:54:53 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 14/43] 202.184.199.34 () {44 vars in 846 bytes} [Fri Mar 13 02:54:56 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 15/44] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:55:00 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21716|app: 0|req: 9/45] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:55:00 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21714|app: 0|req: 11/46] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:55:01 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 0)
[pid: 21714|app: 0|req: 12/47] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:55:01 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21715|app: 0|req: 16/48] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:55:01 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
[pid: 21716|app: 0|req: 10/49] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 02:55:01 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (2 switches on core 1)
[pid: 21715|app: 0|req: 17/50] 202.184.199.34 () {44 vars in 846 bytes} [Fri Mar 13 02:55:04 2020] GET /taoli/ => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 1)
*** Starting uWSGI 2.0.18 (64bit) on [Fri Mar 13 03:00:13 2020] ***
compiled with version: 7.5.0 on 12 March 2020 15:24:34
os: Linux-4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019
nodename: ip-172-31-45-93
machine: x86_64
clock source: unix
detected number of CPU cores: 1
current working directory: /home/ubuntu/arbitrage
writing pidfile to uwsgi.pid
detected binary path: /home/ubuntu/.local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /home/ubuntu/arbitrage
*** WARNING: you are running uWSGI without its master process manager ***
your processes number limit is 3840
your memory page size is 4096 bytes
 *** WARNING: you have enabled harakiri without post buffering. Slow upload could be rejected on post-unbuffered webservers *** 
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to UNIX address /tmp/invest.sock fd 3
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
PEP 405 virtualenv detected: /home/ubuntu/arbitrage/
Set PythonHome to /home/ubuntu/arbitrage/
Python main interpreter initialized at 0x556b076c9900
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 240 seconds
mapped 834320 bytes (814 KB) for 50 cores
*** Operational MODE: preforking+threaded ***
Traceback (most recent call last):
  File "./main_flask.py", line 27, in <module>
    @application.route('/taoli')
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1314, in decorator
    self.add_url_rule(rule, endpoint, f, **options)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 98, in wrapper_func
    return f(self, *args, **kwargs)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1274, in add_url_rule
    rule = self.url_rule_class(rule, methods=methods, **options)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/werkzeug/routing.py", line 666, in __init__
    raise ValueError("urls must start with a leading slash")
ValueError: urls must start with a leading slash
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI worker 1 (pid: 24036, cores: 10)
spawned uWSGI worker 2 (pid: 24038, cores: 10)
spawned uWSGI worker 3 (pid: 24039, cores: 10)
spawned uWSGI worker 4 (pid: 24040, cores: 10)
spawned uWSGI worker 5 (pid: 24041, cores: 10)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/1] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:00:33 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/2] 202.184.199.34 () {46 vars in 870 bytes} [Fri Mar 13 03:00:39 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 1)
Gracefully killing worker 1 (pid: 24036)...
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/3] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:01:26 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 2)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/4] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:01:28 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 3)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/5] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:01:28 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 4)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/6] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:01:29 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 5)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/7] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:01:29 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 6)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/8] 202.184.199.34 () {44 vars in 846 bytes} [Fri Mar 13 03:01:33 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 7)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/9] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:02:54 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 8)
--- no python application found, check your startup logs for errors ---
[pid: 24038|app: -1|req: -1/10] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:02:55 2020] GET /taoli/ => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 9)
*** Starting uWSGI 2.0.18 (64bit) on [Fri Mar 13 03:05:15 2020] ***
compiled with version: 7.5.0 on 12 March 2020 15:24:34
os: Linux-4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019
nodename: ip-172-31-45-93
machine: x86_64
clock source: unix
detected number of CPU cores: 1
current working directory: /home/ubuntu/arbitrage
writing pidfile to uwsgi.pid
detected binary path: /home/ubuntu/.local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /home/ubuntu/arbitrage
*** WARNING: you are running uWSGI without its master process manager ***
your processes number limit is 3840
your memory page size is 4096 bytes
 *** WARNING: you have enabled harakiri without post buffering. Slow upload could be rejected on post-unbuffered webservers *** 
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to UNIX address /tmp/invest.sock fd 3
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
PEP 405 virtualenv detected: /home/ubuntu/arbitrage/
Set PythonHome to /home/ubuntu/arbitrage/
Python main interpreter initialized at 0x556e6048ea50
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 240 seconds
mapped 834320 bytes (814 KB) for 50 cores
*** Operational MODE: preforking+threaded ***
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x556e6048ea50 pid: 24207 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI worker 1 (pid: 24207, cores: 10)
spawned uWSGI worker 2 (pid: 24209, cores: 10)
spawned uWSGI worker 3 (pid: 24210, cores: 10)
spawned uWSGI worker 4 (pid: 24211, cores: 10)
spawned uWSGI worker 5 (pid: 24212, cores: 10)
[pid: 24209|app: 0|req: 1/1] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:05:40 2020] GET /taoli/ => generated 2142 bytes in 26953 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
Gracefully killing worker 1 (pid: 24207)...
[pid: 24209|app: 0|req: 2/2] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:14:56 2020] GET /taoli/ => generated 2142 bytes in 33886 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 1)
Fri Mar 13 03:18:38 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli/ (202.184.199.34)
OSError: write error
[pid: 24209|app: 0|req: 3/3] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 03:17:09 2020] GET /taoli/ => generated 0 bytes in 89269 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 2)
[pid: 24209|app: 0|req: 7/4] 106.39.189.91 () {48 vars in 1057 bytes} [Fri Mar 13 03:29:02 2020] GET /taoli/%E8%BF%99%E9%87%8C%E5%90%97%EF%BC%9F => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 6)
[pid: 24209|app: 0|req: 8/5] 1.68.102.188 () {44 vars in 991 bytes} [Fri Mar 13 03:29:08 2020] GET /taoli/%20%20%E8%BF%99%E9%87%8C%E5%90%97%EF%BC%9F => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 7)
Fri Mar 13 03:29:11 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli/ (49.7.3.95)
OSError: write error
[pid: 24209|app: 0|req: 8/6] 49.7.3.95 () {48 vars in 1003 bytes} [Fri Mar 13 03:28:29 2020] GET /taoli/ => generated 0 bytes in 42287 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 3)
[pid: 24209|app: 0|req: 9/7] 1.68.102.188 () {44 vars in 979 bytes} [Fri Mar 13 03:29:13 2020] GET /taoli/%20%20%E8%BF%99%E9%87%8C%E5%90%97 => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 8)
[pid: 24209|app: 0|req: 10/8] 1.68.102.188 () {44 vars in 967 bytes} [Fri Mar 13 03:29:13 2020] GET /taoli/%20%20%E8%BF%99%E9%87%8C => generated 232 bytes in 0 msecs (HTTP/1.1 404) 2 headers in 87 bytes (1 switches on core 9)
Fri Mar 13 03:29:23 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli/ (49.7.3.95)
OSError: write error
[pid: 24209|app: 0|req: 11/9] 49.7.3.95 () {48 vars in 1003 bytes} [Fri Mar 13 03:28:31 2020] GET /taoli/ => generated 0 bytes in 51851 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 4)
Fri Mar 13 03:29:40 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli/ (49.7.3.95)
OSError: write error
[pid: 24209|app: 0|req: 11/10] 49.7.3.95 () {48 vars in 1003 bytes} [Fri Mar 13 03:28:47 2020] GET /taoli/ => generated 0 bytes in 52959 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 5)
Fri Mar 13 03:30:30 2020 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) !!!
Fri Mar 13 03:30:30 2020 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /taoli/ (1.68.102.188)
OSError: write error
[pid: 24209|app: 0|req: 11/11] 1.68.102.188 () {44 vars in 935 bytes} [Fri Mar 13 03:29:14 2020] GET /taoli/ => generated 0 bytes in 76499 msecs (HTTP/1.1 200) 2 headers in 0 bytes (0 switches on core 0)
[pid: 24209|app: 0|req: 12/12] 66.102.6.141 () {38 vars in 652 bytes} [Fri Mar 13 04:32:19 2020] GET /taoli/ => generated 2453 bytes in 25751 msecs (HTTP/1.1 200) 2 headers in 81 bytes (2 switches on core 1)
[pid: 24209|app: 0|req: 13/13] 202.184.199.34 () {46 vars in 811 bytes} [Fri Mar 13 05:51:15 2020] GET /taoli/ => generated 1514 bytes in 23764 msecs (HTTP/1.1 200) 2 headers in 81 bytes (2 switches on core 2)
[2020-03-13 06:10:18,284] ERROR in app: Exception on /taoli/ [GET]
Traceback (most recent call last):
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "./main_flask.py", line 33, in taoli
    or float(item['cell']['discount_rt'].replace('%','')) <=-6:
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/requests/models.py", line 898, in json
    return complexjson.loads(self.text, **kwargs)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/simplejson/__init__.py", line 525, in loads
    return _default_decoder.decode(s)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/home/ubuntu/arbitrage/lib/python3.6/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
[pid: 24209|app: 0|req: 14/14] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 06:10:17 2020] GET /taoli/ => generated 290 bytes in 731 msecs (HTTP/1.1 500) 2 headers in 99 bytes (2 switches on core 6)
[pid: 24209|app: 0|req: 15/15] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 06:10:20 2020] GET /taoli/ => generated 1806 bytes in 22713 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 7)
[pid: 24209|app: 0|req: 16/16] 202.184.199.34 () {44 vars in 846 bytes} [Fri Mar 13 08:55:14 2020] GET /taoli/ => generated 1808 bytes in 34948 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 3)
*** Starting uWSGI 2.0.18 (64bit) on [Fri Mar 13 09:12:59 2020] ***
compiled with version: 7.5.0 on 12 March 2020 15:24:34
os: Linux-4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019
nodename: ip-172-31-45-93
machine: x86_64
clock source: unix
detected number of CPU cores: 1
current working directory: /home/ubuntu/arbitrage
writing pidfile to uwsgi.pid
detected binary path: /home/ubuntu/.local/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /home/ubuntu/arbitrage
*** WARNING: you are running uWSGI without its master process manager ***
your processes number limit is 3840
your memory page size is 4096 bytes
 *** WARNING: you have enabled harakiri without post buffering. Slow upload could be rejected on post-unbuffered webservers *** 
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to UNIX address /tmp/invest.sock fd 3
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02)  [GCC 8.3.0]
PEP 405 virtualenv detected: /home/ubuntu/arbitrage/
Set PythonHome to /home/ubuntu/arbitrage/
Python main interpreter initialized at 0x55702a5d3900
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 240 seconds
mapped 834320 bytes (814 KB) for 50 cores
*** Operational MODE: preforking+threaded ***
WSGI app 0 (mountpoint='') ready in 0 seconds on interpreter 0x55702a5d3900 pid: 11431 (default app)
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI worker 1 (pid: 11431, cores: 10)
spawned uWSGI worker 2 (pid: 11433, cores: 10)
spawned uWSGI worker 3 (pid: 11434, cores: 10)
spawned uWSGI worker 4 (pid: 11435, cores: 10)
spawned uWSGI worker 5 (pid: 11436, cores: 10)
[pid: 11433|app: 0|req: 1/1] 202.184.199.34 () {46 vars in 877 bytes} [Fri Mar 13 09:13:18 2020] GET /taoli/ => generated 1371 bytes in 28370 msecs (HTTP/1.1 200) 2 headers in 81 bytes (1 switches on core 0)
